src/compiler/emit_x86.bas

```diff
@@ -4153,8 +4153,30 @@ private sub hCMPI _
 		ostr = "test " + dst + COMMA + dst
 		outp ostr
 	else
-		ostr = "cmp " + dst + COMMA + src
-		outp ostr
+
+		'' immediate and offset? it's invalid for CMP so use a temp reg
```

Comment:
Somehow this issue seemed familiar - cc8638c2febc43c86e9a7994a87a4db349728ba6

Reply:
That's perfect.  Almost the same problem.  Will revert emit_x86.bas and let IR handle it.